packages/components/src/resize-tooltip/utils.js

```diff
@@ -0,0 +1,264 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+import useResizeAware from 'react-resize-aware';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useRef, useState } from '@wordpress/element';
+
+const { clearTimeout, setTimeout } = window;
+
+export const POSITIONS = {
+	bottom: 'bottom',
+	cursor: 'cursor',
+	corner: 'corner',
+};
+
+/**
+ * @typedef {Object} UseResizeLabelProps
+ *
+ * @property {boolean} isActive The visibility of the label.
+ * @property {undefined|string} label The label value.
+ * @property {Function} resizeListener Element to be rendered for resize listening events.
+ */
+
+/**
+ * Custom hook that manages resize listener events. It also provides a label
+ * based on current resize width x height values.
+ *
+ * @param {Object} props
+ * @param {string} props.axis Only shows the label corresponding to the axis.
+ * @param {number} props.fadeTimeout Duration (ms) before deactivating the resize label.
+ * @param {Function} props.onMove Callback when a resize occurs. Provides onMouseEvent event callback.
+ * @param {boolean} props.onResize Callback when a resize occurs. Provides { width, height } callback.
+ * @param {string} props.position Adjusts label value.
+ * @param {boolean} props.showPx Whether to add `PX` to the label.
+ *
+ * @return {UseResizeLabelProps} Properties for hook.
+ */
+export function useResizeLabel( {
+	axis,
+	fadeTimeout = 180,
+	onMove = noop,
+	onResize = noop,
+	position = POSITIONS.cursor,
+	showPx = false,
+} ) {
+	const [ isDragging, setIsDragging ] = useState( false );
+	/**
```

Comment:
Our situation in WP-JS is a little awkward when it comes to multi-line comments, but in any case [we should avoid the leading double asterisk](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/#multi-line-comments) unless it's opening a JSDoc block.

Reply:
Oop! 🙊 . Thank you for the reminder! This is something I have to get use to!